src/Mobs/Monster.h

```diff
@@ -74,6 +76,32 @@ class cMonster :
 	virtual void CheckEventSeePlayer(cChunk & a_Chunk);
 	virtual void EventSeePlayer(cPlayer * a_Player, cChunk & a_Chunk);
 
+	// tolua_begin
+	/** Returns whether the mob can be leashed. */
+	bool CanBeLeashed() const { return m_CanBeLeashed; }
+
+	/** Sets whether the mob can be leashed, for extensibility in plugins */
+	void SetCanBeLeashed(bool a_CanBeLeashed) { m_CanBeLeashed = a_CanBeLeashed; }
+
+	/** Returns whether the monster is leashed to an entity. */
+	bool IsLeashed() const { return (m_LeashedTo != nullptr); }
+
+	/** Leash the monster to an entity. */
+	void LeashTo(cEntity * a_Entity, bool a_ShouldBroadcast = true);
+
+	/** Unleash the monster. */
+	void Unleash(bool a_ShouldDropLeashPickup, bool a_ShouldBroadcast = true, bool a_ShouldRemoveFromEntity = true);
```

Comment:
The function has three params, so the Lua binding accepts three params, too. Does it make sense for all three of them to be accessible to plugins? And their documentation is missing.

If you don't want to write manual bindings for this, just add an overload that has less params and calls this internally, and make it the only exported API function.